### DISCUSSION TITLE
Use of netstat over reading /proc files

### DISCUSSION THREAD

**Discussion by jake-at-work** [oshi/oshi#2433] [Q&A] [answered]
I am curious if there has ever been any discussion about optimizing the Linux network metrics by utilizing the various /proc files rather than executing `netstat` and parsing its output? It looks like for the most part on Linux that `netstat` is parsing the /proc files itself and just reformatting it. The execution of `netstat` results in a noticeable latency in sampling these metrics that is not seem direct parsing of /proc files. 

If it is just a matter of lack of contribution I would love to contribute in any way but don't want to tread on something that may be intentionally implemented in a particular way.

**Comment by dbwiddis** [comment, marked as answer]
Wow, sorry I missed seeing this discussion for a year.  Probably should have been an issue. :|

I'd totally welcome a performance improvement.

**Reply by jake-at-work** [reply]
It has only taken me 5 months to see your reply. No problem. I went back and forth on it being an issue or not. I might be picking this up soon.
